spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -422,15 +422,32 @@
       end
     end
 
+    describe "retirement income - line 20a" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:show_retirement_ui).and_return(true)
```

Comment:
[boulder] - I believe the `show_retirement_ui` flag being `false` is the current state for users, so we'd want it covered by tests to make sure nothing is leaking out.

Reply:
My rationale behind not testing the disabled flag state was that it felt like testing internal behavior of the flag rather than desired behavior of our code, but as I'm typing that out, I feel like my `if ... && enabled...` is in fact something OUR code does. Will add!

Reply:
Do you think we should go back and add flag-off tests to the #5529 controller as well?

Reply:
oh good idea, thanks for adding that one too 👍